opentelemetry/src/common.rs

```diff
@@ -402,82 +402,55 @@ pub trait ExportError: std::error::Error + Send + Sync + 'static {
 
 /// Information about a library or crate providing instrumentation.
 ///
-/// An instrumentation library should be named to follow any naming conventions
+/// An instrumentation scope should be named to follow any naming conventions
 /// of the instrumented library (e.g. 'middleware' for a web framework).
 ///
 /// See the [instrumentation libraries] spec for more information.
 ///
 /// [instrumentation libraries]: https://github.com/open-telemetry/opentelemetry-specification/blob/v1.9.0/specification/overview.md#instrumentation-libraries
 #[derive(Debug, Default, Clone)]
 #[non_exhaustive]
-pub struct InstrumentationLibrary {
+pub struct InstrumentationScope {
     /// The library name.
     ///
     /// This should be the name of the crate providing the instrumentation.
     pub name: Cow<'static, str>,
 
     /// The library version.
-    ///
-    /// # Examples
-    ///
-    /// ```
-    /// let library = opentelemetry::InstrumentationLibrary::builder("my-crate").
-    ///     with_version(env!("CARGO_PKG_VERSION")).
-    ///     with_schema_url("https://opentelemetry.io/schemas/1.17.0").
-    ///     build();
-    /// ```
     pub version: Option<Cow<'static, str>>,
 
-    /// [Schema url] used by this library.
+    /// [Schema URL] used by this library.
     ///
-    /// [Schema url]: https://github.com/open-telemetry/opentelemetry-specification/blob/v1.9.0/specification/schemas/overview.md#schema-url
+    /// [Schema URL]: https://github.com/open-telemetry/opentelemetry-specification/blob/v1.9.0/specification/schemas/overview.md#schema-url
     pub schema_url: Option<Cow<'static, str>>,
 
     /// Specifies the instrumentation scope attributes to associate with emitted telemetry.
-    pub attributes: Vec<KeyValue>,
+    pub attributes: Cow<'static, Vec<KeyValue>>,
 }
 
-// Uniqueness for InstrumentationLibrary/InstrumentationScope does not depend on attributes
-impl Eq for InstrumentationLibrary {}
+// Uniqueness for InstrumentationScope/InstrumentationScope does not depend on attributes
```

Comment:
there was a recent clarification that scope attributes also contribute to uniqueness test.
https://github.com/open-telemetry/opentelemetry-specification/blob/main/CHANGELOG.md#logs-1

Let fix in a follow up PR to keep this PR's scope.

Reply:
Tracked via https://github.com/open-telemetry/opentelemetry-rust/issues/2225